avro-converter/src/main/java/io/confluent/connect/avro/AvroData.java

```diff
@@ -1402,14 +1509,14 @@ else if (schema.type() == Schema.Type.INT32)
   private static String unionMemberFieldName(org.apache.avro.Schema schema) {
     if (schema.getType() == org.apache.avro.Schema.Type.RECORD ||
         schema.getType() == org.apache.avro.Schema.Type.ENUM) {
-      return splitName(schema.getName())[1];
+      return schema.getFullName();
     }
     return schema.getType().getName();
   }
 
   private static String unionMemberFieldName(Schema schema) {
     if (schema.type() == Schema.Type.STRUCT || isEnumSchema(schema)) {
-      return splitName(schema.name())[1];
+      return schema.name();
```

Comment:
Agreed, same response also, will apply the same solution to what we agree on that one.